src/components/DetailPage.vue

```diff
@@ -0,0 +1,125 @@
+<script setup lang="ts">
+import axios from "axios";
+import { useRoute, useRouter } from "vue-router";
+import { onMounted, ref, computed } from "vue";
+import {required, helpers} from '@vuelidate/validators'
+import {useVuelidate} from '@vuelidate/core'
+
+const includeVue = (name:string) => {return name.toLowerCase().includes('vue')}
+const alphaNumSpace = helpers.regex(/^[A-Za-z0-9 ]*$/)
+
+
+const rules = {
+    todoName: {required,
+      alphaNumSpace: helpers.withMessage('Only letters, numbers, and spaces are allowed', alphaNumSpace),
+      includeVue: helpers.withMessage('Must include vue in title', includeVue)},
+}
+
+const data = ref({
+    todoName: '',
+    isComplete: false
+})
+
+const v$ = useVuelidate(rules, data)
+
+type Post = {
+  _id: string;
+  isComplete: boolean;
+  todoName: string;
+};
+
+const route = useRoute();
+const router = useRouter()
+const id = ref();
+const posts = ref<Post[]>([])
+const apiURL = "https://calm-plum-jaguar-tutu.cyclic.app/";
+
+function retrievePosts() {
+  axios
+    .get(apiURL + "todos")
+    .then((result) => {
+      posts.value = result.data.data;
+      console.log(posts.value);
+    })
+    .catch((error) => {
+      console.log("error", error);
+    });
+}
+
+function editTitle() {
+  // actually deletes original post and makes a new post
+  v$.value.$touch()
+  if (!v$.value.$invalid) {
+    axios.delete(apiURL + `todos/${post.value?._id}`)
+    .then((result) => {
+      console.log(result);
+    })
+    .catch((error) => {
+      console.log("delete error", error);
+    })
+    .then(() => {
+      return axios.post(apiURL + "todos", { todoName: data.value.todoName , isComplete: data.value.isComplete});
+    }).then((result) => {
+        console.log(result)
+        router.push('/')
+        }).catch((error) => {
+            console.log("post error", error)
+        });
+  }
+}
+
+onMounted(() => {
+  retrievePosts();
+  id.value = route.params.id;
+});
+
+const post = computed(() => {
+  return posts.value.find((item) => {
+    return item._id === id.value;
+  });
+});
+
+const validPost = computed(() => {
+  if (post !== undefined) {
+    return true;
+  } else {
+    return false;
+  }
```

Comment:
You're missing the `.value` on the post variable here and additionally your if statements could be simplified to `const validPost = computed(() => post.value !== undefined`